### PR TITLE
Addition of DomainTag to required functions for the creation of metrics required for Domain Cost Attribution

### PR DESCRIPTION
**Pull request by abhishekj720**
**What changed?**

- Modified file common/persistence/elasticsearch/esVisibilityMetricClients.go
- Added Scope parameter in UpdateErrorMetric function.
- Changed IncCounter function by adding scope type variable to get domain tag.

**Why?**

- To create metrics based on DomainId to get the domain cost attribution of each customer.

**How did you test it?**

- Tested locally. Reviewed from Zijian.

**Potential risks**

- Error in metrics creation.

**Release notes**
- No, migration and updates are not required.

**Documentation Changes**

### CODE REVIEW

**Review by Shaddoll — COMMENTED** [review]
test comment

File: common/metrics/client.go [hunk 1]
```diff
@@ -70,13 +70,17 @@ func NewClient(scope tally.Scope, serviceIdx ServiceIdx) Client {
...
+
 	name := string(m.metricDefs[counterIdx].metricName)
 	m.childScopes[scopeIdx].Counter(name).Inc(1)
 }
 
 // AddCounter adds delta to the counter and
 // emits to the metrics backend
 func (m *ClientImpl) AddCounter(scopeIdx int, counterIdx int, delta int64) {
+	//need to add DomainId/Name parameter
+
```
Comment (Shaddoll) [inline common/metrics/client.go:83]: test comment

File: common/persistence/elasticsearch/esVisibilityMetricClients.go [hunk 2]
```diff
@@ -55,14 +55,17 @@ func (p *visibilityMetricsClient) RecordWorkflowExecutionStarted(
...
+	
+	sw := scopeWithDomainTag.StartTimer(metrics.ElasticsearchLatency)
 
-	sw := p.metricClient.StartTimer(metrics.ElasticsearchRecordWorkflowExecutionStartedScope, metrics.ElasticsearchLatency)
 	err := p.persistence.RecordWorkflowExecutionStarted(ctx, request)
 	sw.Stop()
 
 	if err != nil {
-		p.updateErrorMetric(metrics.ElasticsearchRecordWorkflowExecutionStartedScope, err)
+		p.updateErrorMetric(scopeWithDomainTag,metrics.ElasticsearchRecordWorkflowExecutionStartedScope, err)
```
Comment (Shaddoll) [inline common/persistence/elasticsearch/esVisibilityMetricClients.go:68]: nit: need a space after `,`. Could you run `make lint` to have the linter fix this?

File: common/metrics/client.go [hunk 3]
```diff
@@ -70,13 +70,17 @@ func NewClient(scope tally.Scope, serviceIdx ServiceIdx) Client {
 // IncCounter increments one for a counter and emits
 // to metrics backend
 func (m *ClientImpl) IncCounter(scopeIdx int, counterIdx int) {
+	//need to add DomainId/Name parameter
```
Comment (Shaddoll) [inline common/metrics/client.go:73]: nit: remove this comment

File: common/metrics/client.go [hunk 4]
```diff
@@ -70,13 +70,17 @@ func NewClient(scope tally.Scope, serviceIdx ServiceIdx) Client {
...
+
 	name := string(m.metricDefs[counterIdx].metricName)
 	m.childScopes[scopeIdx].Counter(name).Inc(1)
 }
 
 // AddCounter adds delta to the counter and
 // emits to the metrics backend
 func (m *ClientImpl) AddCounter(scopeIdx int, counterIdx int, delta int64) {
+	//need to add DomainId/Name parameter
```
Comment (Shaddoll) [inline common/metrics/client.go:82]: nit: remove this comment

File: README.md [hunk 5]
```diff
@@ -1,4 +1,6 @@
 # Cadence
+
```
Comment (Shaddoll) [inline README.md:2]: nit: remove the changes in this file

File: common/metrics/client.go [hunk 6]
```diff
@@ -113,6 +117,8 @@ func (m *ClientImpl) Scope(scopeIdx int, tags ...Tag) Scope {
 	scope := m.childScopes[scopeIdx]
 	return newMetricsScope(scope, scope, m.metricDefs, false).Tagged(tags...)
 }
+//add domainID to the tags
```
Comment (Shaddoll) [inline common/metrics/client.go:120]: nit: remove the comment

File: common/persistence/elasticsearch/esVisibilityMetricClients.go [hunk 7]
```diff
@@ -293,30 +317,36 @@ func (p *visibilityMetricsClient) DeleteWorkflowExecution(
...
-		p.updateErrorMetric(metrics.ElasticsearchDeleteWorkflowExecutionsScope, err)
+		p.updateErrorMetric(scopeWithDomainTag, metrics.ElasticsearchDeleteWorkflowExecutionsScope, err)
 	}
 
 	return err
 }
 
-func (p *visibilityMetricsClient) updateErrorMetric(scope int, err error) {
+func (p *visibilityMetricsClient) updateErrorMetric(scopeWithDomainTag metrics.Scope, scope int, err error) {
+												//scopeIdx			Tags ...tags
```
Comment (Shaddoll) [inline common/persistence/elasticsearch/esVisibilityMetricClients.go:335]: nit: remove these comments

**Review by Shaddoll — APPROVED** [review]

File: common/persistence/elasticsearch/esVisibilityMetricClients.go [hunk 8]
```diff
@@ -293,30 +317,33 @@ func (p *visibilityMetricsClient) DeleteWorkflowExecution(
 	ctx context.Context,
 	request *p.VisibilityDeleteWorkflowExecutionRequest,
 ) error {
-	p.metricClient.IncCounter(metrics.ElasticsearchDeleteWorkflowExecutionsScope, metrics.ElasticsearchRequests)
 
-	sw := p.metricClient.StartTimer(metrics.ElasticsearchDeleteWorkflowExecutionsScope, metrics.ElasticsearchLatency)
+	scopeWithDomainTag := p.metricClient.Scope(metrics.ElasticsearchDeleteWorkflowExecutionsScope, metrics.DomainTag(request.Domain))
```
Comment (Shaddoll) [inline common/persistence/elasticsearch/esVisibilityMetricClients.go:321]: VisibilityDeleteWorkflowExecutionRequest doesn't have the Domain field

**Review by Shaddoll — CHANGES_REQUESTED** [review]
VisibilityDeleteWorkflowExecutionRequest doesn't have the Domain field. Could you please fix the issue?

File: common/persistence/serialization/interfaces.go [hunk 9]
```diff
@@ -285,6 +285,7 @@ type (
 	// TimerTaskInfo blob in a serialization agnostic format
 	TimerTaskInfo struct {
 		DomainID        UUID
+		Domain          string
```
Comment (Shaddoll) [inline common/persistence/serialization/interfaces.go:288]: Could you revert this change? This is not necessary

Comment (abhishekj720) [reply common/persistence/serialization/interfaces.go:288]: Done. Do you want me to add any other change as well before i push it again?

**Review by Shaddoll — APPROVED** [review]
